### PR TITLE
feat(layers): transport layers

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Implementating a simple `Transport` logging for `RpcClient`. 

## Solution

Implement a custom `Transport` layer called `LoggingLayer` which logs requests being sent and incoming responses.

## PR Checklist

- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
Is this something we want to include in alloy directly, basically the foundry retry implementation?

**Comment by frisitano** [comment]
> Is this something we want to include in alloy directly, basically the foundry retry implementation?

It would be ideal if this could be integrated into `alloy` directly in such a way that allows for a generic retry policy to be provided. I think a lot of consumers currently have to implement the full stack in their own codebase such as what I did [here](https://github.com/fractal-zkp/zero-bin/blob/tx-proof-rpc-new/rpc/src/retry.rs), which is undesirable.

File: examples/layers/examples/simple_retry.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        let inner = self.inner.clone();
+        let policy = self.policy.clone();
+
+        let mut inner = std::mem::replace(&mut self.inner, inner);
+        Box::pin(async move {
+            let mut retries = 0;
+
+            let mut res = inner.call(req.clone()).await;
+
+            while let Some(_policy) = policy.retry(&req, res.as_ref()) {
```
Comment (frisitano) [inline examples/layers/examples/simple_retry.rs:146]: I think the policy should be updated here such that any changes to the policy on an iteration take affect. See [here](https://github.com/fractal-zkp/zero-bin/blob/2cbcf98ea8bf57ee17775d2e72c51dd40e8cd04b/rpc/src/retry.rs#L119-L135) for an example.

File: README.md [hunk 2]
```diff
@@ -40,6 +40,9 @@ This repository contains the following examples:
   - [x] [Nonce management filler](./examples/fillers/examples/nonce_filler.rs)
   - [x] [Recommended fillers](./examples/fillers/examples/recommended_fillers.rs)
   - [x] [Signer management filler](./examples/fillers/examples/signer_filler.rs)
+- [x] Layers
+  - [x] [Req/Res Logging Layer](./examples/layers/examples/logging.rs)
```
Comment (zerosnacks) [inline README.md:44]: Prefer using the full name

**Comment by gakonst** [comment]
Should we close this given we're doing the retry layer upstream?

**Comment by yash-atreya** [comment]
> Should we close this given we're doing the retry layer upstream?

I'd 

> Should we close this given we're doing the retry layer upstream?

I have removed the retry layer implementation, I'd still like to include the LoggingLayer, to show what needs to be done to implemented your own custom layer.

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -39,6 +39,7 @@ jobs:
             -e 'subscribe_pending_transactions' \
             -e 'trace_call' \
             -e 'trace_transaction' \
+            -e 'simple_retry' \
```
Comment (zerosnacks) [inline .github/workflows/integration.yml:42]: Should be removed

File: Cargo.toml [hunk 4]
```diff
@@ -88,6 +88,9 @@ alloy = { version = "0.1", features = [
     "node-bindings",
     "rpc-types-debug",
     "rpc-types-trace",
+    "rpc-types",
+    "json-rpc",
+    "rpc-client",
```
Comment (zerosnacks) [inline Cargo.toml:93]: `rpc-types` should not be necessary as it is enabled by any `rpc-types-*` flag automatically

File: README.md [hunk 5]
```diff
@@ -39,7 +39,11 @@ This repository contains the following examples:
   - [x] [Gas estimation filler](./examples/fillers/examples/gas_filler.rs)
   - [x] [Nonce management filler](./examples/fillers/examples/nonce_filler.rs)
   - [x] [Recommended fillers](./examples/fillers/examples/recommended_fillers.rs)
+  - [x] [Signer management filler](./examples/fillers/examples/signer_filler.rs)
   - [x] [Wallet management filler](./examples/fillers/examples/wallet_filler.rs)
+- [x] Layers
+  - [x] [Request/Response Logging Layer](./examples/layers/examples/logging.rs)
+  - [x] [Simple Retry Layer](./examples/layers/examples/simple_retry.rs)
```
Comment (zerosnacks) [inline README.md:46]: Remove `signer_filler` ref, superseded by `wallet_filler` + remove redundant `simple_retry` link

**Pull request merged**